vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/credentialverifier/validator/MsoMdocValidator.kt

```diff
@@ -33,18 +38,29 @@ class MsoMdocValidator {
                 logger.severe("validUntil / validFrom is not available in the credential's MSO")
                 throw ValidationException(ERROR_MESSAGE_INVALID_DATE_MSO, ERROR_CODE_INVALID_DATE_MSO)
             }
-            val isCurrentTimeGreaterThanValidFrom =
-                DateUtils.isDatePassedCurrentDate(validFrom.toString())
-            val isCurrentTimeLessThanValidUntil =
-                !DateUtils.isDatePassedCurrentDate(validUntil.toString())
+            val isValidFromIsFutureDate =
+                DateUtils.isFutureDateWithTolerance(validFrom.toString())
```

Comment:
validFrom attribute is expected to be before current DateTime, the function looks like we are checking if validFrom is future date (validFrom < currentTime). Is this expected here?

Reply:
Yes, validFrom should come before current date, so when validFrom is Future date , we are throwing exception.
```
            if(isValidFromIsFutureDate){
                logger.severe("Error while doing validity verification - invalid validFrom in the MSO of the credential")
                throw ValidationException(ERROR_MESSAGE_INVALID_VALID_FROM_MSO, ERROR_CODE_INVALID_VALID_FROM_MSO)
            }
```